plugins/woocommerce/includes/class-woocommerce.php

```diff
@@ -1038,6 +1039,43 @@ public function initialize_session() {
 		}
 	}
 
+	/**
+	 * Tell bots not to index some WooCommerce-created directories.
+	 *
+	 * We try to detect the default "User-agent: *" added by WordPress and add our rules to that group, because
+	 * it's possible that some bots will only interpret the first group of rules if there are multiple groups with
+	 * the same user agent.
+	 *
+	 * @param string $output The contents that WordPress will output in a robots.txt file.
+	 *
+	 * @return string
+	 */
+	public function robots_txt( $output ) {
+		$path = ( ! empty( $site_url['path'] ) ) ? $site_url['path'] : '';
+
+		$lines       = explode( PHP_EOL, $output );
```

Comment:
This assumes that the already existing output is using the system's end of line convention to separate lines, which may not be the case. In fact, when running `wp eval 'echo do_robots();'` in Windows the output is

```
User-agent: *
Disallow: /wordpress/wp-admin/
Allow: /wordpress/wp-admin/admin-ajax.php


User-agent: *
Disallow: /wp-content/uploads/wc-logs/
Disallow: /wp-content/uploads/woocommerce_transient_files/
Disallow: /wp-content/uploads/woocommerce_uploads/
```

instead of

```
User-agent: *
Disallow: /wp-content/uploads/wc-logs/
Disallow: /wp-content/uploads/woocommerce_transient_files/
Disallow: /wp-content/uploads/woocommerce_uploads/
Disallow: /wp-admin/
Allow: /wp-admin/admin-ajax.php
```

which is the expected value (and what gets correctly generated in Linux). Somewhere there's some code that is blindly using `\n` to separate the lines - and that's fine, [the robots.txt specification](https://developers.google.com/search/docs/crawling-indexing/robots/robots_txt#file-format) allows any kind of line separators to be used.

The solution is to change the `explode` to the following, which covers all the possible cases:

```suggestion
		$lines       = preg_split('/\r\n|\r|\n/', $output );
```

Reply:
Good call. I'm glad that you got picked for this review so you could lend some Windows perspective to this! 😄